tests/schema/tests.py

```diff
@@ -1832,3 +1833,67 @@ def test_alter_pk_with_self_referential_field(self):
         new_field.set_attributes_from_name('id')
         with connection.schema_editor() as editor:
             editor.alter_field(Node, old_field, new_field)
+
+    @mock.patch('django.db.backends.base.schema.datetime')
+    @mock.patch('django.db.backends.base.schema.timezone')
+    def test_add_datefield_and_datetimefield_use_effective_default(self, mocked_datetime,
+                                                                   mocked_tz):
+        """
+        #25005 - effective_default() should be used for DateField,
+        DateTimeField and TimeField if auto_now or auto_add_now is set.
+        """
+        def assertDefault(field, field_name, expected_default):
+            with connection.cursor() as cursor:
+                editor.add_field(Author, field)
+                cursor.execute("SELECT %s FROM schema_author;" % field_name)
+                self.assertEqual(cursor.fetchall()[0][0], expected_default)
```

Comment:
I'd prefer something like `check_added_field_default()`; but I'd really rather see it defined as an independent method (or even function), rather than an inner function. It seems like a useful general utility for schema tests.